drivers/photo-electronic/adpd410x/adpd410x.c

```diff
@@ -0,0 +1,660 @@
+
+	/* Set precondition PD */
+	ret = adpd410x_reg_read(dev, (ADPD410X_REG_CATHODE_A + timeslot_no * 0x20),
+				&data);
+	if(ret != SUCCESS)
+		return FAILURE;
+	data &= ~BITM_CATHODE_A_PRECON_A;
+	data |= (init->precon_option << BITP_CATHODE_A_PRECON_A) &
+		BITM_CATHODE_A_PRECON_A;
+	ret = adpd410x_reg_write(dev, (ADPD410X_REG_CATHODE_A + timeslot_no * 0x20),
+				 data);
+	if(ret != SUCCESS)
+		return FAILURE;
+
+	/**
+	 *  Set TIA VREF and TRIM options. The 0xE000 is writing reserved bits
+	 *  as specified in the datasheet.
+	 */
+	data = init->afe_trim_opt << BITP_AFE_TRIM_A_AFE_TRIM_VREF_A |
+	       init->vref_pulse_opt << BITP_AFE_TRIM_A_VREF_PULSE_VAL_A |
+	       init->chan2 << BITP_AFE_TRIM_A_TIA_GAIN_CH2_A |
+	       init->chan1 << BITP_AFE_TRIM_A_TIA_GAIN_CH1_A;
+	ret = adpd410x_reg_write(dev,
+				 (ADPD410X_REG_AFE_TRIM_A + timeslot_no * 0x20), data);
+	if(ret != SUCCESS)
+		return FAILURE;
+
+	/* Set LED pattern */
+	data = init->pulse4_subtract << BITP_PATTERN_A_SUBTRACT_A |
+	       init->pulse4_reverse << BITP_PATTERN_A_REVERSE_INTEG_A;
+	ret = adpd410x_reg_write(dev, (ADPD410X_REG_PATTERN_A + timeslot_no * 0x20),
+				 data);
+	if(ret != SUCCESS)
+		return FAILURE;
+
+	/* Set bytes number for time slot */
+	data = (init->byte_no << BITP_DATA1_A_SIGNAL_SIZE_A) &
+	       BITM_DATA1_A_SIGNAL_SIZE_A;
+	ret = adpd410x_reg_write(dev, (ADPD410X_REG_DATA1_A + timeslot_no * 0x20),
+				 data);
+	if(ret != SUCCESS)
+		return FAILURE;
+
+	/* Set decimate factor */
+	data = (init->dec_factor << BITP_DECIMATE_A_DECIMATE_FACTOR_A) &
+	       BITM_DECIMATE_A_DECIMATE_FACTOR_A;
+	ret = adpd410x_reg_write(dev,
+				 (ADPD410X_REG_DECIMATE_A + timeslot_no * 0x20), data);
+	if(ret != SUCCESS)
+		return FAILURE;
+
+	/* Set LED power */
+	data = init->led1.value |
+	       (init->led2.value << BITP_LED_POW12_A_LED_CURRENT2_A);
+	ret = adpd410x_reg_write(dev,
+				 (ADPD410X_REG_LED_POW12_A + timeslot_no * 0x20), data);
+	if(ret != SUCCESS)
+		return FAILURE;
+	data = init->led3.value |
+	       (init->led4.value << BITP_LED_POW34_A_LED_CURRENT4_A);
+	ret = adpd410x_reg_write(dev,
+				 (ADPD410X_REG_LED_POW34_A + timeslot_no * 0x20), data);
+	if(ret != SUCCESS)
+		return FAILURE;
+
+	/* Set ADC cycle and repeat */
+	if(init->adc_cycles == 0)
+		init->adc_cycles = 1;
+	if(init->repeats_no == 0)
+		init->repeats_no = 1;
+	data = init->repeats_no | (init->adc_cycles << BITP_COUNTS_A_NUM_INT_A);
+	return adpd410x_reg_write(dev, (ADPD410X_REG_COUNTS_A + timeslot_no * 0x20),
+				  data);
+}
+
+/**
+ * @brief Get number of bytes in the device FIFO.
+ * @param dev - Device handler.
+ * @param bytes - Pointer to the byte count container.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t adpd410x_get_fifo_bytecount(struct adpd410x_dev *dev, uint16_t *bytes)
+{
+	int32_t ret;
+
+	ret = adpd410x_reg_read(dev, ADPD410X_REG_FIFO_STATUS, bytes);
+	if(ret != SUCCESS)
+		return FAILURE;
+
+	*bytes &= BITM_INT_STATUS_FIFO_FIFO_BYTE_COUNT;
```

Comment:
I think bytes parameter could be better described. I don't get if it's a count (number) or a special bit field (because you do masking here).

Reply:
Should I name it differently? The comments should explain what it is.